src/main/java/org/datadog/jenkins/plugins/datadog/publishers/DatadogQueuePublisher.java

```diff
@@ -76,8 +76,14 @@ protected void doRun() throws Exception {
             long pending = queue.getPendingItems().size();
             long stuck = 0;
             long blocked = 0;
+            
             final Queue.Item[] items = queue.getItems();
             for (Queue.Item item : items) {
+                // Get the name of the task for job_name tag
+                Queue.Task task = item.task;
+                String display_name = task.getFullDisplayName();
```

Comment:
I don't think `task.getFullDisplayName` is the right tool here. I tried it for a pipeline job and it returns `part of Pipeline #19` which is an unbounded tag.
Also Task in an interface so any plugin is free to implement getFullDisplayName the way it wants. I believe we have to limit ourselves to known implementations of that interface.

What do you think of something like this? This has been slightly tested but it may not be the perfect solution to use as is.

```java
                Queue.Task task = item.task;
                if(task instanceof FreeStyleProject){
                    // Normal jenkins project
                    logger.severe("job name is: " + task.getFullDisplayName());
                } else if (task instanceof ExecutorStepExecution.PlaceholderTask) {
                    // Pipeline task
                      Run<?, ?> run = ((PlaceholderTask) task).runForDisplay();
                      if(run != null) {
                          logger.severe(job name is: " + run.getParent().getFullName());
                      }
                }
```

Reply:
I originally had the test `(task instanceof Job)`, but took it out for simplicity. This makes more sense, though and I'll look into doing it similar to this way.

Reply:
This way requires the workflow plugin- would it make sense to still check if the task is a job, but support pipeline tasks in the future when we add those dependencies?

Reply:
> This way requires the workflow plugin- would it make sense to still check if the task is a job, but support pipeline tasks in the future when we add those dependencies?

Yes let's add a TODO for this :)

Reply:
Updated to check for `Run`, which covers pipelines, then default to display name. https://github.com/jenkinsci/datadog-plugin/pull/73/files#diff-bfee0dd89969cc21252e65dc11b29ef0R89